File: inciport-website/src/models/IncidentReport.ts
import ContactInformation from './ContactInformation';
import Location from './Location';
import ChosenMainCategory from './ChosenMainCategory';
import WorkerTeam from './WorkerTeam';

export default class IncidentReport {
	id: number;
	status: string;
	chosenMainCategory: ChosenMainCategory;
	location: Location;
	description: string;
	contactInformation: ContactInformation;
	imageUrls: string[];
	timestampCreatedUtc: Date;
	timestampModifiedUtc: Date;
	assignedTeam: WorkerTeam;

	constructor(
		id: number,
		status: string,
		chosenMainCategory: ChosenMainCategory,
		location: Location,
		description: string,
		contactInformation: ContactInformation,
		imageUrls: string[],
		timestampCreatedUtc: Date,
		timestampModifiedUtc: Date,
		assignedTeam: WorkerTeam
	) {
		this.id = id;
		this.status = status;
		this.chosenMainCategory = chosenMainCategory;
		this.location = location;
		this.description = description;
		this.contactInformation = contactInformation;
		this.imageUrls = imageUrls;
		this.timestampCreatedUtc = timestampCreatedUtc;
		this.timestampModifiedUtc = timestampModifiedUtc;
		this.assignedTeam = assignedTeam;
	}
}

File: inciport-webservice/InciportWebService.Domain/Models/Municipality/IncidentReport/IncidentReport.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace InciportWebService.Domain {

  public class IncidentReport {
    public int Id { get; set; }

    public ReportStatus Status { get; set; }
    public virtual ChosenMainCategory ChosenMainCategory { get; set; }
    public virtual Location Location { get; set; }
    public string Description { get; set; }
    public virtual ContactInformation ContactInformation { get; set; }
    public virtual WorkerTeam AssignedTeam { get; set; }
    public virtual List<ImageReference> ImageReferences { get; set; }
    public DateTimeOffset TimestampCreatedUtc { get; set; }
    public DateTimeOffset TimestampLastModifiedUtc { get; set; }

    public IncidentReport() {
      ImageReferences = new List<ImageReference>();
    }
  }
}